### PR TITLE
getty-generator: add kernel cmdline and env vars to disable it

### PR DESCRIPTION
**Pull request by bluca**
systemd.getty_auto/rd.systemd.getty_auto/SYSTEMD_GETTY_AUTO can be used
to disable the generator. Enabled by default.

### CODE REVIEW

File: man/systemd-getty-generator.xml [hunk 1]
```diff
@@ -53,6 +53,38 @@
...
+  <refsect1>
+    <title>Kernel Command Line</title>
+
+    <para><filename>systemd-getty-generator</filename> understands the following kernel command line
+    parameters:</para>
+
+    <variablelist class='kernel-commandline-options'>
+      <varlistentry>
+        <term><varname>systemd.getty_auto</varname></term>
+        <term><varname>rd.systemd.getty_auto</varname></term>
```
Comment (poettering) [inline man/systemd-getty-generator.xml]: please also add reference to kernel-command-line(7)

File: man/systemd-getty-generator.xml [hunk 2]
```diff
@@ -53,6 +53,38 @@
...
+
```
Comment (poettering) [inline man/systemd-getty-generator.xml:66]: the `rd.` version should go: we don't do gettys in the initrd anyway...

File: man/systemd-getty-generator.xml (hunk 1 above)
Comment (poettering) [inline man/systemd-getty-generator.xml]: please suffix with "=", since we expect an argument

File: man/systemd-getty-generator.xml [hunk 3]
```diff
@@ -53,6 +53,38 @@
...
+        <listitem><para>Those options take an optional boolean argument, and default to yes.
```
Comment (poettering) [inline man/systemd-getty-generator.xml]: Those → These?

File: man/systemd-getty-generator.xml [hunk 4]
```diff
@@ -53,6 +53,38 @@
...
+        The generator is enabled by default, and a negative value may be used to disable it.
```
Comment (poettering) [inline man/systemd-getty-generator.xml]: negative → false?

(i.e. most people would probably set =1 and =0 as values, but if you write negative people might misunderstand tht this means =-1...)

File: src/getty-generator/getty-generator.c [hunk 5]
```diff
@@ -139,11 +142,50 @@ static int run_container(void) {
         }
 }
 
+static int parse_proc_cmdline_item(const char *key, const char *value, void *data) {
+        int r;
+
+        assert(key);
+
+        if (proc_cmdline_key_streq(key, "systemd.getty_auto") ||
+            (proc_cmdline_key_streq(key, "rd.systemd.getty_auto") && in_initrd())) {
```
Comment (poettering) [inline src/getty-generator/getty-generator.c]: as mentioned, please drop the rd. flavour

**Review by poettering — CHANGES_REQUESTED** [review]
looks good, but some fixes please

**Comment by bluca** [comment]
updated

**Comment by poettering** [comment]
kernel-command-line mn page fix is still missing

**Comment by bluca** [comment]
> kernel-command-line mn page fix is still missing

I added it - or did you mean at the bottom?

**Comment by bluca** [comment]
added in "See also" as well

**Comment by poettering** [comment]
> > kernel-command-line mn page fix is still missing
> 
> I added it - or did you mean at the bottom?

man/kernel-command-line.xml is still missing an entry for this!

**Comment by bluca** [comment]
> > > kernel-command-line mn page fix is still missing
> > 
> > 
> > I added it - or did you mean at the bottom?
> 
> man/kernel-command-line.xml is still missing an entry for this!

ahah sorry, I thought you meant adding a link from this manpage to that one - will send another PR

**Pull request merged**